src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs

```diff
@@ -129,7 +129,12 @@ private void OutMember(Expression instance, MemberInfo member, Type exprType)
 
             string name = null;
             if (_conventions.PropertyNameConverter != null)
-                name = _conventions.PropertyNameConverter(member);
+            {   
+                //do not use convention for types in system namespaces
+                if(member.DeclaringType?.Namespace?.StartsWith("System") == false &&
+                   member.DeclaringType?.Namespace?.StartsWith("Microsoft") == false)
+                    name = _conventions.PropertyNameConverter(member);
```

Comment:
@myarichuk Do we need to apply this in PropertyNameConventionJSMetadataProvider also?

Reply:
Yeah, I think we should apply it there too.